static_frame/core/util.py

```diff
@@ -3187,15 +3187,27 @@ def iloc_to_insertion_iloc(key: int, size: int) -> int:
         raise IndexError(f'index {key} out of range for length {size} container.')
     return key % size
 
-def total_getsizeof(iter: tp.Iterable[any]) -> int:
+def total_getsizeof(it: tp.Iterable[any], *, seen=None) -> int:
     '''
     Gives the total size of an iterable of elements
-    '''
+    see also: https://code.activestate.com/recipes/577504/
+    '''
+    handlers = {
+        tuple: iter,
+        list: iter,
+        dict: lambda d: chain.from_iterable(d.items()),
+        set: iter,
+        frozenset: iter,
+        FrozenAutoMap: iter
+    }
+    seen = set() if seen is None else seen
     total = 0
-    seen = set()
-    for e in iter:
-        if id(e) in seen:
+    for el in it:
+        if id(el) in seen:
             continue
-        seen.add(id(e))
-        total += getsizeof(e)
+        seen.add(id(el))
+        total += getsizeof(el)
+        for t, h in handlers.items():
+            if isinstance(el, t):
```

Comment:
I feel like this approach is bit to heavy... we do not need to try every element in `dict`, we just need to handle the one special case: `dict`. I think better is not define `handlers`, and instead just branch here when you find a `dict`.

Reply:
While `dict` is the special case, elements of types not included handlers must not be looped on (e.g. `int`, `str`, `FrozenAutoMap`). So we need a branch here based on the element's type being in the handlers and then special handling based on if the element is a dict/ndarray.

Reply:
Ah, I see... maybe you just need to instead look for non-string iterable objects, i.e.: `hasattr(obj, '__len__') and not isinstance(obj, str)`. Calling `isinstance` many times for every object might get costly.

Reply:
Unfortunately there are a couple iterable objects that we don't want to loop over (like `FrozenAutoMap`). I think running a `hasattr(obj, '__len__')` could be a good thing to check before going to a bunch of `isinstance` calls if we are worried about performance cost.

Reply:
Interesting... why do you not want to count the keys in a `FrozenAutoMap`?